cohttp-async/src/cohttp_async.ml

```diff
@@ -1,11 +1,8 @@
 module Body = Body
 module Body_raw = Body [@@deprecated "Use Body"]
 module Client = Client
-module Io = Io [@@deprecated "This module is not for public consumption"]
+module Io = Io.IO
```

Comment:
Why removed this deprecation?

Reply:
We forward the new IO type (reader + buffer) in the expert response callback. This IO module will be the only way to perform any read operations there so leaving it deprecated doesn’t seem ideal.

Reply:
I thought about recreating a new reader and forward that to the expert callback, but that has an edge case where if the user decides to use the expert mode for any reason other than connection upgrades (to websocket etc) then eventually the control will come back to Cohttp and we’ll need to continue processing more requests and would need to recreate the new input channel again.

Reply:
I brought back the deprecation. I moved to a different approach so instead of making the entire IO module public, I just moved a restricted Input_channel module within the server module as its only useful there when working with expert responses.